src/TCommandLine.cpp

```diff
@@ -524,20 +516,40 @@ void TCommandLine::adjustHeight()
 
 void TCommandLine::spellCheck()
 {
-    if (!mpHost->mEnableSpellCheck) {
+    if (!mpHost || !mpHost->mEnableSpellCheck) {
         return;
     }
 
     QTextCursor oldCursor = textCursor();
     QTextCharFormat f;
     QTextCursor c = textCursor();
     c.select(QTextCursor::WordUnderCursor);
-    QByteArray encodedText = mpHunspellCodec->fromUnicode(c.selectedText());
-    if (!Hunspell_spell(mpHunspell, encodedText.constData())) {
-        // Word is misspelt
-        f.setUnderlineStyle(QTextCharFormat::SpellCheckUnderline);
-        f.setUnderlineColor(Qt::red);
-        f.setFontUnderline(true);
+    QByteArray encodedText = mpHost->mpConsole->getHunspellCodec_system()->fromUnicode(c.selectedText());
+    if (!Hunspell_spell(mpHost->mpConsole->getHunspellHandle_system(), encodedText.constData())) {
+        // Word is not in selected system dictionary
+        Hunhandle* userDictionaryhandle = mpHost->mpConsole->getHunspellHandle_user();
+        if (userDictionaryhandle) {
+            if (Hunspell_spell(userDictionaryhandle, c.selectedText().toUtf8().constData())) {
+                // We are using a user dictionary and it does contain this word - so
+                // use a different underline, on many systems the spell-check underline is
+                // a wavy line but on macOs it is a dotted line - so use dash underline
+                f.setUnderlineStyle(QTextCharFormat::DashUnderline);
+                f.setUnderlineColor(Qt::cyan);
```

Comment:
Well ***I*** want to know if it is a word I have okayed (i.e. is something that has been added from a MUD or personal whim) or it has come from the main dictionary - if it is the **former** then I can change it - and I can tell that without having to right click and wait (there is a little delay remember to do the additional `Huspell_suggest(...)` calls.  I would hesitate to say I can make it optional but that is quite *do-abIe* if you cannot agree to this.

The choice of making it a dashed underline cyan was so that it was a *reassuring* greenish colour and clearly not the style of a spelling error (but not so green that it would be a problem for red-green colour bindness - although a bright blue would be better that would not contrast with the selection blue background; the different underline style was also chosen as a fallback if the colour was not enough).

Reply:
What do others on team think?